src/borg/testsuite/archiver.py

```diff
@@ -4204,36 +4204,36 @@ def test_format_option(self):
 
         self.create_regular_file('a_file', size=8)
         test_archive0 = self.repository_location + '::test0'
-        self.cmd('create', test_archive0, self.input_path)
+        self.cmd('create', test_archive0, "input")#self.input_path)
 
         os.unlink('input/a_file')
         self.create_regular_file('a_file', size=33)
         atime, mtime = 123457520, 234900290
         os.utime('input/a_file', (atime, mtime))
 
         test_archive1 = self.repository_location + '::test1'
-        self.cmd('create', test_archive1, self.input_path)
+        self.cmd('create', test_archive1, "input")#self.input_path)
 
-        output_file_mod = self.cmd('diff', test_archive0, "test1")
-        output2 = self.cmd(
+        output_default = self.cmd('diff', test_archive0, "test1")
+        output_default_explicit_fmt = self.cmd(
             'diff', '--format',
             '{newsize:8d} B {oldsize:8d} B {path}{NEWLINE}',
             test_archive0, "test1")
-        output3 = self.cmd(
+        expected_default = [
+            "    +33 B      -8 B input/a_file"
+        ]
+        self.assert_equal(output_default, output_default_explicit_fmt)
+        self.assert_equal(expected_default, output_default_explicit_fmt.splitlines())
+
+        output_nontrivial_fmt1 = self.cmd(
             'diff', '--format',
             '{change} ({deltasize:8d} B) {mtime:%s} {path}{NL}',
```

Comment:
8d - hmm, is this really an integer that gets formatted?

iirc it could be a string, see that "+" in there, for a normal int, that is not present.

Reply:
can you check?

Reply:
Well, in the test case it is an integer. The idea is to allow the user to provide a formatting, right?
This is a case that should fail without this feature and should pass once the feature is implemented.
Do I get it wrong? How houls I test this?

Reply:
but just "8d" won't generate output like in line 4223 (the normal integer formatting does not output a "+" for positive numbers).

but the implementation for `--format` is missing still, so you'll notice that later anyway.

Reply:
Yes, the implementation is missing: my bad. Sorry.
But you are right I would have notice it later.
And yes, the test is wrong: it should read `{deltasize:8d}` or change the next line to `modified (      33 B) 234900290 input/a_file`...